Encountify/Services/LocationDatabaseAccess.cs

```diff
@@ -0,0 +1,84 @@
+﻿using Encountify.Models;
+using SQLite;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+
+namespace Encountify.Services
+{
+    //it may can be made generic for all databases and or merged with IDataStore
+    //however it might become a huge hussle if we when adding databases with different logic
+    class LocationDatabaseAccess
+    {
+        public List<Location> GetLocationList()
+        {
+            string dbPath = Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.Personal), 
+                DatabaseAccessConstants.LocationDatabaseName);
+            SQLiteConnection db = new SQLiteConnection(dbPath);
+            //haven't tested it, however we may not need to create new SQLiteConnection in every request
```

Comment:
You're right. It's not necessary if we don't use db.close();

Reply:
Could you link the source that says that it is the only time that new connection must be generated. As far I have tested it does work flawlessly, though if I remove it and some operations does invalidate SQLiteConnection, then we might run into some very strange and hard to debug issues later on.

Reply:
![image](https://archive-media-0.nyafuu.org/vp/image/1551/30/1551300060713.png)